### ISSUE TITLE
"int not callable" Exception when retry value passed to datastore operation

### ISSUE THREAD

**Issue by keegan2149** [googleapis/python-api-core#143]
Thanks for stopping by to let us know something could be better!

**PLEASE READ**: If you have a support contract with Google, please create an issue in the [support console](https://cloud.google.com/support/) instead of filing on GitHub. This will ensure a timely response.

Please run down the following list and make sure you've tried the usual "quick fixes":

  - Search the issues already opened: https://github.com/googleapis/python-api-core/issues
  - Search StackOverflow: https://stackoverflow.com/questions/tagged/google-cloud-platform+python

If you are still having issues, please be sure to include as much information as possible:

#### Environment details

  - OS type and version: Macos 11.1
  - Python version: python 3.7.9
  - pip version: pip 21.0.1
  - `google-api-core` version: 
  - Name: google-api-core
Version: 1.23.0
Summary: Google API client core library
Home-page: https://github.com/googleapis/python-api-core
Author: Google LLC
Author-email: googleapis-packages@google.com
License: Apache 2.0
Location: .../.../.../.../env/lib/python3.7/site-packages
Requires: pytz, protobuf, setuptools, google-auth, six, googleapis-common-protos, requests
Required-by: google-cloud-logging, google-cloud-firestore, google-cloud-datastore, google-cloud-core, google-api-python-client

#### Steps to reproduce

  1. Create datastore entities list
  2. Call put_multi and pass a non-zero retry value

#### Code example

```
    retry = 3
    timeout = 1000
    
    logger.debug('updating database')
    _,timestamp = get_timestamp()
    start_timestamp = timestamp
    result = client.put_multi(entities, retry=retry, timeout=timeout)
```

#### Stack trace
```
'int' object is not callable
```

Making sure to follow these steps will guarantee the quickest resolution possible.

Thanks!

**Comment by busunkim96** [comment]
Hi @keegan2149,

The datastore library expects a retry object:
https://github.com/googleapis/python-datastore/blob/bf1dde60b2f42e939c7dfa4a5228c3f41d565ece/google/cloud/datastore/client.py#L582-L587

Please see https://googleapis.dev/python/google-api-core/latest/retry.html for documentation on how to construct a `Retry` object.

**Comment by keegan2149** [comment]
Any idea why its not applied by default? It looks like the retry object was removed along with the import statements.  I ran some test operations and it looks pretty stable.

**Comment by busunkim96** [comment]
@keegan2149 I don't quite follow, are you saying the datastore client code was recently changed? As far as I know the cloud clients have always expected full `Retry` objects.

Clients *will* accept bare timeouts. A '3' gets translated into a timeout of 3s. I believe we don't do this with the Retry object because it's not as clear which of the items below should end up being set to 3. In the worst case, it might end up giving folks the false impression that the request will be retried 3 times.

- initial (float) – The minimum amount of time to delay in seconds. This must be greater than 0.
- maximum (float) – The maximum amount of time to delay in seconds.
- multiplier (float) – The multiplier applied to the delay.
- deadline (float) – How long to keep retrying in seconds. The last sleep period is shortened as necessary, so that the last retry runs at deadline (and not considerably beyond it).

**Comment by busunkim96** [comment]
Closing due to inactivity. If you're still experiencing this, please open a new issue.
